libs/s3/s3_versioning_common_test_lib.py

```diff
@@ -20,41 +20,343 @@
+                                  bucket_name: str = None,
+                                  object_name: str = None,
+                                  **kwargs) -> None:
+    """
+    Verify GET/HEAD Object response for specified version/object
+
+    :param s3_ver_test_obj: S3VersioningTestLib object to perform S3 versioning calls
+    :param version_id: Optional version ID for GET/HEAD Object call.
+        In case it is not specified, object is retrieved instead of a specific version.
+    :param bucket_name: Target bucket name
+    :param object_name: Target object name
+    :param **kwargs: Optional keyword arguments
+        "etag": Expected ETag of the version/object
+        "get_error_msg": Error message to verify for GET Object
+        "head_error_msg": Error message to verify for HEAD Object
+    """
+    etag = kwargs.get("etag", None)
+    get_error_msg = kwargs.get("get_error_msg", None)
+    head_error_msg = kwargs.get("head_error_msg", None)
+    LOG.info("Verifying GET Object with VersionId response")
+    try:
+        if version_id:
+            get_response = s3_ver_test_obj.get_object_version(bucket_name, object_name,
+                                                              version_id=version_id)
+        else:
+            get_response = s3_ver_test_obj.get_object(bucket=bucket_name, key=object_name)
+        assert_utils.assert_true(get_response[0], get_response[1])
+        if version_id:
+            assert_utils.assert_equal(get_response[1]["ResponseMetadata"]["VersionId"],
+                                      version_id)
+        if etag:
+            assert_utils.assert_equal(get_response[1]["ResponseMetadata"]["ETag"], etag)
+        LOG.info("Successfully performed GET Object: %s", get_response)
+    except CTException as error:
+        LOG.error(error.message)
+        if not error_msg:
+            raise CTException(err.CLI_ERROR, error.args[0]) from error
+        assert_utils.assert_in(get_error_msg, error.message, error.message)
+    LOG.info("Verifying HEAD Object with VersionId response")
+    try:
+        if version_id:
+            head_response = s3_ver_test_obj.head_object_version(bucket=bucket_name,
+                                                                key=object_name,
+                                                                version_id=version_id)
+        else:
+            head_response = s3_ver_test_obj.object_info(bucket_name=bucket_name,
+                                                        key=object_name)
+        assert_utils.assert_true(head_response[0], head_response[1])
+        if version_id:
+            assert_utils.assert_equal(
+                head_response[1]["ResponseMetadata"]["VersionId"], version_id)
+        if etag:
+            assert_utils.assert_equal(head_response[1]["ResponseMetadata"]["ETag"], etag)
+        LOG.info("Successfully performed HEAD Object: %s", head_response)
+    except CTException as error:
+        LOG.error(error.message)
```

Comment:
Do you need exception handling in these functions?

Reply:
Even when require and not in this particular case, please log traceback which adds more readable logging.

Reply:
Exception handling needed to assert error messages. Calls to logger updated to log the traceback.